src/main/decompInitMod.F90

```diff
@@ -306,6 +256,164 @@ subroutine decompInit_lnd(lni, lnj, amask)
     end if
     call shr_sys_flush(iulog)
 
+  !------------------------------------------------------------------------------
+  ! Internal subroutines for this subroutine
```

Comment:
The purpose of using internal subroutines is so that all of the declarations at the top of decompInit_lnd are assumed in the internal subroutines below. This makes it an easier transition to having these calls here. It also makes them private and only usable within decompInit_lnd.